controller/src/main/java/org/jboss/as/controller/OperationContextImpl.java

```diff
@@ -2524,4 +2540,38 @@ public int hashCode() {
         }
     }
 
+    private static class HostCapabilityContext implements CapabilityContext {
+        @Override
+        public boolean canSatisfyRequirements(CapabilityContext dependentContext) {
+            return true;
```

Comment:
This should be:

```
return dependentContext == CapabilityContext.GLOBAL || dependentContext instanceof HostCapabilityContext;
```

Otherwise capabilities from domain level profile and socket-binding-group resources will be able to find requirements from the host context.